# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/RenderCompositorD3D11SWGL.h
# Commit: 91d8f0f829fd
# Full Hash: 91d8f0f829fd2ae6ec234a6a3f372bf503ed2fbd
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2020-11-06 04:14:34
# Regressor Bug: 1673983
# File Overlap Count: 2
# Description:
#   Bug 1673983 - Add RenderTextureHostSWGL support to D3D11 RenderTextureHosts. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D95799
# ==============================================================================

diff -r 91cb8ee8ddd8 -r 91d8f0f829fd gfx/webrender_bindings/RenderCompositorD3D11SWGL.h
--- a/gfx/webrender_bindings/RenderCompositorD3D11SWGL.h	Thu Nov 05 19:26:21 2020 +0000
+++ b/gfx/webrender_bindings/RenderCompositorD3D11SWGL.h	Thu Nov 05 20:18:13 2020 +0000
@@ -49,6 +49,7 @@
   layers::WebRenderCompositor CompositorType() const override {
     return layers::WebRenderCompositor::D3D11;
   }
+  RenderCompositorD3D11SWGL* AsRenderCompositorD3D11SWGL() { return this; }
 
   // Interface for wr::Compositor
   CompositorCapabilities GetCompositorCapabilities() override;
@@ -95,6 +96,8 @@
     int32_t mY;
   };
 
+  ID3D11Device* GetDevice() { return mCompositor->GetDevice(); }
+
  private:
   RefPtr<layers::CompositorD3D11> mCompositor;
   void* mContext = nullptr;